### PR TITLE
Fix build warnings

### LINKED ISSUE #306: Build warnings
Shouldn't be too hard to fix.  
The deprecations require some refactoring into new framework functions.

```
w: file:///[...]gotify/login/LoginActivity.kt:155:13 'startActivityForResult(Intent, Int): Unit' is deprecated. Deprecated in Java
w: file:///[...]gotify/login/LoginActivity.kt:166:15 'onActivityResult(Int, Int, Intent?): Unit' is deprecated. Overrides deprecated member in 'androidx.activity.ComponentActivity'. Deprecated in Java
w: file:///[...]gotify/messages/MessagesActivity.kt:249:19 'onBackPressed(): Unit' is deprecated. Overrides deprecated member in 'androidx.core.app.ComponentActivity'. Deprecated in Java
w: file:///[...]gotify/messages/MessagesModelFactory.kt:12:75 Unchecked cast: MessagesModel! to T
w: file:///[...]gotify/service/WebSocketConnection.kt:204:51 'getter for activeNetworkInfo: NetworkInfo?' is deprecated. Deprecated in Java
w: file:///[...]gotify/service/WebSocketConnection.kt:205:49 'getter for isConnected: Boolean' is deprecated. Deprecated in Java
w: file:///[...]gotify/settings/SettingsActivity.kt:113:28 'setTargetFragment(Fragment?, Int): Unit' is deprecated. Deprecated in Java
```

```
/home/runner/[...]res/values/strings.xml:39:4: Multiple substitutions specified in non-positional format of string resource string/other_error. Did you mean to add the formatted="false" attribute?
/home/runner/[...]res/values/strings.xml:5:4: Multiple substitutions specified in non-positional format of string resource string/version_failed. Did you mean to add the formatted="false" attribute?
/home/runner/[...]res/values/strings.xml:61:4: Multiple substitutions specified in non-positional format of string resource string/version_failed_status_code. Did you mean to add the formatted="false" attribute?
/home/runner/[...]res/values/strings.xml:66:4: Multiple substitutions specified in non-positional format of string resource string/versions. Did you mean to add the formatted="false" attribute?
```

### PR DESCRIPTION
**Pull request by cyb3rko**
Warnings listed in #306 are fixed.  
I've never used a certificate for Gotify, but at least choosing one from your phone and loading / parsing it does work.

Closes #306

### CODE REVIEW

**Comment by cyb3rko** [comment]
Just in case you wonder why only 3 commits are signed:  
I've moved to Linux and forgot to enable auto-signing.

File: app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt [hunk 1]
```diff
@@ -242,12 +249,15 @@ internal class MessagesActivity :
...
 
-    override fun onBackPressed() {
-        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
-            binding.drawerLayout.closeDrawer(GravityCompat.START)
-        } else {
-            super.onBackPressed()
+    private fun addBackPressCallback() {
+        onBackPressedCallback = object : OnBackPressedCallback(false) {
+            override fun handleOnBackPressed() {
+                if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
```
Comment (jmattheis) [inline app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt:255]: Do we need this check here? I'd expect that the callback is only executed when it's enabled, and it's only enabled if the drawer is open.

Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt:255]: I don't exactly know. If in any circumstances it is called and the drawer is not open, would that matter?  
I suppose it doesn't throw an exception, at least according to the docs:  
https://developer.android.com/reference/androidx/drawerlayout/widget/DrawerLayout#closeDrawer(int)

If so, we don't need that here, true.

Comment (jmattheis) [reply app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt:255]: yeah, the two extra lines don't really hurt the code-base, so it's probably not a big deal if they stay and aren't really needed.

**Review by jmattheis — APPROVED** [review]

**Pull request merged**